File: CS50/Python/readability/readability.py
"""
 readability.py
 Program that computes the approximate grade level needed to comprehend some text, per the below.
 By Pol
 Created 23/04/2022
"""

def main():
    # Prompting the user for a text
    txt = input("Text: ")

    # Counting the words in the Text
    words = txt.split()
    words = len(words)

    # Counting the letters in the Text
    letters = count_let(txt)

    # Counting the sentences in the Text
    sentences = count_sen(txt)

    # Determinating the texts' grade with the Coleman-Liau index
    avg_let = float((letters / words) * 100)
    avg_sen = float((sentences / words) * 100)
    index = 0.0588 * avg_let - 0.296 * avg_sen - 15.8
    grade = round(index)

    # Printing the grade
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print("Grade", int(grade))


# _________________o____________________
# Function that counts letters in a text
def count_let(txt):
    count = 0
    for word in txt:
        for letter in word:
            if letter.isalpha() == True:
                count += 1
    return count

# _________________o____________________
# Function that counts sentences in a text
def count_sen(txt):
    count = 0
    for word in txt:
        for letter in word:
            if letter in (".", "!", "?"):
                count += 1
    return count


# _________________o____________________
# Calling main
main()

File: CS50/C/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

// Declaring the functions
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    // Storing each count in seperate variables
    int letters   = count_letters(text);
    int words     = count_words(text);
    int sentences = count_sentences(text);

    // Transforming values to doubles to do the calculus
    double letters1 = letters;
    double words1 = words;
    double sentences1 = sentences;

    // Doing the "redability test" with "Coleman-Liau"'s index formula
    double L = (letters1 / words1) * 100;
    double S = (sentences1 / words1) * 100;
    double grade1 = (0.0588 * L) - (0.296 * S) - 15.8;
    int grade = round(grade1);

    // Printing the text's grade-reading level
    if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}


// Function that counts letters
int count_letters(string text)
{
    int len = strlen(text);
    int l_count = 0;
    for (int i = 0; i < len; i++)
    {
        // Verifying if the char is a letter
        if (isalpha(text[i]) != 0)
        {
            l_count++;
        }
    }
    return l_count;
}

// Function that counts words
int count_words(string text)
{
    int len = strlen(text);
    int w_count = 1;
    for (int i = 0; i < len; i++)
    {
        // Verifying if the char is an space
        if (isspace(text[i]) != 0)
        {
            w_count++;
        }
    }
    return w_count;
}

// Function that counts senteces
int count_sentences(string text)
{
    int len = strlen(text);
    int s_count = 0;
    for (int i = 0; i < len; i++)
    {
        // Verifying if the char is a sentence
        if ((text[i] == '.') || (text[i] == '!') || (text[i] == '?'))
        {
            s_count++;
        }
    }
    return s_count;
}
